src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.cpp

```diff
@@ -438,9 +440,95 @@ bool ReconstructionEngine_panorama::Adjust()
       return false;
   }
 
+  //Assuming only a small number of views have outliers due to motion
+  if (cleanWithPriors())
+  {
+    // Minimize Rotation
+    success = BA.adjust(_sfmData, BundleAdjustmentPanoramaCeres::REFINE_ROTATION);
+    if(success)
+    {
+        ALICEVISION_LOG_INFO("Bundle successfully refined: Rotation after cleaning outliers");
+    }
+    else
+    {
+        ALICEVISION_LOG_INFO("Failed to refine: Rotation");
+        return false;
+    }
+  }
+
   return true;
 }
 
+bool ReconstructionEngine_panorama::cleanWithPriors()
+{
+    if (_rotationPriors.size() != _sfmData.getViews().size())
+    {
+        return false;
+    }
+
+    //Remove all matches in sfm
+    sfm::Constraints2D & constraints2d = _sfmData.getConstraints2D();
+    constraints2d.clear();
+
+    //Put back the rotations
+    _sfmData.getPoses() = _rotationPriors;
```

Comment:
I would move that in the caller function, and keep this function create matches with a custom maxAngle that can be called in the 2 cases.

Reply:
To me, that's not the purpose of this PR which was only to enforce XML priors. Filtering matches to handle more outliers seems a large problem which is out of topic here.